warehouse/integrations/bigquery/bigquery.go

```diff
@@ -137,54 +139,44 @@ func New(conf *config.Config, log logger.Logger) *BigQuery {
 	return bq
 }
 
-func (bq *BigQuery) getMiddleware() *middleware.Client {
-	if bq.middleware != nil {
-		return bq.middleware
-	}
-	return middleware.New(
-		bq.db,
-		middleware.WithLogger(bq.logger),
-		middleware.WithKeyAndValues(
-			logfield.SourceID, bq.warehouse.Source.ID,
-			logfield.SourceType, bq.warehouse.Source.SourceDefinition.Name,
-			logfield.DestinationID, bq.warehouse.Destination.ID,
-			logfield.DestinationType, bq.warehouse.Destination.DestinationDefinition.Name,
-			logfield.WorkspaceID, bq.warehouse.WorkspaceID,
-			logfield.Schema, bq.namespace,
-		),
-		middleware.WithSlowQueryThreshold(bq.config.slowQueryThreshold),
-	)
-}
-
 func getTableSchema(tableSchema model.TableSchema) []*bigquery.FieldSchema {
 	return lo.MapToSlice(tableSchema, func(columnName, columnType string) *bigquery.FieldSchema {
 		return &bigquery.FieldSchema{Name: columnName, Type: dataTypesMap[columnType]}
 	})
 }
 
+func (bq *BigQuery) partitionColumn(workspaceID string) string {
+	configKey := "Warehouse.bigquery.partitionColumn." + workspaceID
+	return bq.conf.GetString(configKey, "")
+}
+
 func (bq *BigQuery) DeleteTable(ctx context.Context, tableName string) (err error) {
 	tableRef := bq.db.Dataset(bq.namespace).Table(tableName)
 	err = tableRef.Delete(ctx)
 	return
 }
 
 func (bq *BigQuery) CreateTable(ctx context.Context, tableName string, columnMap model.TableSchema) error {
-	bq.logger.Infof("BQ: Creating table: %s in bigquery dataset: %s in project: %s", tableName, bq.namespace, bq.projectID)
+	bq.logger.Infon("Creating table",
+		logger.NewStringField(logfield.ProjectID, bq.projectID),
+		obskit.Namespace(bq.namespace),
+		logger.NewStringField(logfield.TableName, tableName),
+	)
 	sampleSchema := getTableSchema(columnMap)
 	metaData := &bigquery.TableMetadata{
-		Schema:           sampleSchema,
-		TimePartitioning: &bigquery.TimePartitioning{},
+		Schema: sampleSchema,
+		TimePartitioning: &bigquery.TimePartitioning{
+			Type: bigquery.DayPartitioningType,
```

Comment:
Specifying the partition instead of relying on the default value set by the SDK. It is more understandable now, from the code what kind of partition type we do.